From: Michele Locati <michele@locati.it>
Date: Fri, 10 May 2019 08:52:51 +0200
Subject: [PATCH] libtool: hide "Could not determine host file name"

--- a/build-aux/ltmain.sh
+++ b/build-aux/ltmain.sh
@@ -2872,9 +2872,9 @@ func_convert_file_check ()
   $debug_cmd
 
   if test -z "$2" && test -n "$1"; then
-    func_error "Could not determine host file name corresponding to"
-    func_error "  '$1'"
-    func_error "Continuing, but uninstalled executables may not work."
+    #func_error "Could not determine host file name corresponding to"
+    #func_error "  '$1'"
+    #func_error "Continuing, but uninstalled executables may not work."
     # Fallback:
     func_to_host_file_result=$1
   fi
@@ -2891,9 +2891,9 @@ func_convert_path_check ()
   $debug_cmd
 
   if test -z "$4" && test -n "$3"; then
-    func_error "Could not determine the host path corresponding to"
-    func_error "  '$3'"
-    func_error "Continuing, but uninstalled executables may not work."
+    #func_error "Could not determine the host path corresponding to"
+    #func_error "  '$3'"
+    #func_error "Continuing, but uninstalled executables may not work."
     # Fallback.  This is a deliberately simplistic "conversion" and
     # should not be "improved".  See libtool.info.
     if test "x$1" != "x$2"; then

--- a/libcharset/build-aux/ltmain.sh
+++ b/libcharset/build-aux/ltmain.sh
@@ -2872,9 +2872,9 @@ func_convert_file_check ()
   $debug_cmd
 
   if test -z "$2" && test -n "$1"; then
-    func_error "Could not determine host file name corresponding to"
-    func_error "  '$1'"
-    func_error "Continuing, but uninstalled executables may not work."
+    #func_error "Could not determine host file name corresponding to"
+    #func_error "  '$1'"
+    #func_error "Continuing, but uninstalled executables may not work."
     # Fallback:
     func_to_host_file_result=$1
   fi
@@ -2891,9 +2891,9 @@ func_convert_path_check ()
   $debug_cmd
 
   if test -z "$4" && test -n "$3"; then
-    func_error "Could not determine the host path corresponding to"
-    func_error "  '$3'"
-    func_error "Continuing, but uninstalled executables may not work."
+    #func_error "Could not determine the host path corresponding to"
+    #func_error "  '$3'"
+    #func_error "Continuing, but uninstalled executables may not work."
     # Fallback.  This is a deliberately simplistic "conversion" and
     # should not be "improved".  See libtool.info.
     if test "x$1" != "x$2"; then
