From: Michele Locati <michele@locati.it>
Date: Fri, 10 May 2019 09:28:34 +0200
Subject: [PATCH] libtool: hide "library was moved"

--- a/build-aux/ltmain.sh
+++ b/build-aux/ltmain.sh
@@ -7958,7 +7958,7 @@ func_mode_link ()
 	# Find the relevant object directory and library name.
 	if test yes = "$installed"; then
 	  if test ! -f "$lt_sysroot$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library '$lib' was moved."
+	    #func_warning "library '$lib' was moved."
 	    dir=$ladir
 	    absdir=$abs_ladir
 	    libdir=$abs_ladir

--- a/libcharset/build-aux/ltmain.sh
+++ b/libcharset/build-aux/ltmain.sh
@@ -7956,7 +7956,7 @@ func_mode_link ()
 	# Find the relevant object directory and library name.
 	if test yes = "$installed"; then
 	  if test ! -f "$lt_sysroot$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library '$lib' was moved."
+	    #func_warning "library '$lib' was moved."
 	    dir=$ladir
 	    absdir=$abs_ladir
 	    libdir=$abs_ladir
