From: Michele Locati <michele@locati.it>
Date: Fri, 10 May 2019 09:34:07 +0200
Subject: [PATCH] libtool: hide "has not been installed in"

--- a/build-aux/ltmain.sh
+++ b/build-aux/ltmain.sh
@@ -4460,7 +4460,7 @@ func_mode_install ()
 	    fi
 	    libfile=$libdir/`$ECHO "$lib" | $SED 's%^.*/%%g'`
 	    if test -n "$libdir" && test ! -f "$libfile"; then
-	      func_warning "'$lib' has not been installed in '$libdir'"
+	      #func_warning "'$lib' has not been installed in '$libdir'"
 	      finalize=false
 	    fi
 	  done

--- a/libcharset/build-aux/ltmain.sh
+++ b/libcharset/build-aux/ltmain.sh
@@ -4458,7 +4458,7 @@ func_mode_install ()
 	    fi
 	    libfile=$libdir/`$ECHO "$lib" | $SED 's%^.*/%%g'`
 	    if test -n "$libdir" && test ! -f "$libfile"; then
-	      func_warning "'$lib' has not been installed in '$libdir'"
+	      #func_warning "'$lib' has not been installed in '$libdir'"
 	      finalize=false
 	    fi
 	  done
