From 49cd8bde058eb96f0b67fe76287bb637db8f2cc5 Mon Sep 17 00:00:00 2001
From: Bruno Haible <bruno@clisp.org>
Date: Tue, 23 Jan 2018 09:02:02 +0100
Subject: [PATCH 8/9] Don't use an undocumented Autoconf macro.

---
 libcharset/ChangeLog    | 5 +++++
 libcharset/configure.ac | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/libcharset/ChangeLog b/libcharset/ChangeLog
index b6736d5..a2983a5 100644
--- a/libcharset/ChangeLog
+++ b/libcharset/ChangeLog
@@ -1,3 +1,8 @@
+2018-01-23  Bruno Haible  <bruno@clisp.org>
+
+	Don't use an undocumented Autoconf macro.
+	* configure.ac: Use AC_CONFIG_HEADERS instead of AC_CONFIG_HEADER.
+
 2017-07-15  Bruno Haible  <bruno@clisp.org>
 
 	Get rid of autom4te.cache directories, as far as possible.
diff --git a/libcharset/configure.ac b/libcharset/configure.ac
index ef3167f..4a4fb9e 100644
--- a/libcharset/configure.ac
+++ b/libcharset/configure.ac
@@ -1,4 +1,4 @@
-dnl Copyright (C) 2000-2003, 2005-2009, 2016 Free Software Foundation, Inc.
+dnl Copyright (C) 2000-2003, 2005-2009, 2016, 2018 Free Software Foundation, Inc.
 dnl This file is part of the GNU CHARSET Library.
 dnl
 dnl The GNU CHARSET Library is free software; you can redistribute it and/or
@@ -19,7 +19,7 @@ AC_PREREQ([2.60])
 AC_INIT([libcharset], [1.4])
 AC_CONFIG_SRCDIR([lib/localcharset.c])
 AC_CONFIG_AUX_DIR([build-aux])
-AC_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AC_PROG_MAKE_SET
 
 dnl           checks for basic programs
-- 
2.16.2.windows.1

