From: Michele Locati <michele@locati.it>
Date: Tue, 14 May 2019 18:46:19 +0200
Subject: [PATCH] Reduce dependency on libwinpthread

--- a/gettext-tools/config.h
+++ b/gettext-tools/config.h
@@ -696,7 +696,7 @@
 #define HAVE_CHSIZE 1

 /* Define to 1 if you have the `clock_gettime' function. */
-#define HAVE_CLOCK_GETTIME 1
+#define HAVE_CLOCK_GETTIME 0

 /* Define to 1 if you have the `clock_settime' function. */
 #define HAVE_CLOCK_SETTIME 1
