--- a/gettext-tools/src/msginit.c	Fri Jul 08 12:08:39 2016
+++ b/gettext-tools/src/msginit.c	Fri Jul 08 12:07:27 2016
@@ -1140,26 +1140,31 @@
       return plural_table[i].value;
 
   gettextcldrdir = getenv ("GETTEXTCLDRDIR");
   if (gettextcldrdir != NULL && gettextcldrdir[0] != '\0')
     {
-      const char *gettextlibdir;
+      char *gettextbindir;
       char *dirs[3], *last_dir;
       char *argv[4];
       pid_t child;
       int fd[1];
       FILE *fp;
       char *line;
       size_t linesize;
       size_t linelen;
       int exitstatus;
 
-      gettextlibdir = getenv ("GETTEXTLIBDIR");
-      if (gettextlibdir == NULL || gettextlibdir[0] == '\0')
-        gettextlibdir = relocate (LIBDIR "/gettext");
+      gettextbindir = xstrdup(program_name);
+      for (i = strlen(gettextbindir) - 1; i >= 0; i--) {
+        if (gettextbindir[i] == '/' || gettextbindir[i] == '\\') {
+          gettextbindir[i + 1] = '\0';
+          break;
+        }
+      }
 
-      prog = xconcatenated_filename (gettextlibdir, "cldr-plurals", NULL);
+      prog = xconcatenated_filename (gettextbindir, "cldr-plurals.exe", NULL);
+      free(gettextbindir);
 
       last_dir = xstrdup (gettextcldrdir);
       dirs[0] = "common";
       dirs[1] = "supplemental";
       dirs[2] = "plurals.xml";
