From: Bruno Haible <bruno@clisp.org>
Date: Fri, 13 Jun 2025 12:27:44 +0200
Subject: stat: Support the file name "/dev/null" on native Windows

diff --git a/gettext-runtime/gnulib-lib/stat.c b/gettext-runtime/gnulib-lib/stat.c
index ebed63e..6663783 100644
--- a/gettext-runtime/gnulib-lib/stat.c
+++ b/gettext-runtime/gnulib-lib/stat.c
@@ -118,6 +118,10 @@ rpl_stat (char const *name, struct stat *buf)
      around length limitations
      <https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file> ?  */
 
+  /* To ease portability.  Like in open.c.  */
+  if (strcmp (name, "/dev/null") == 0)
+    name = "NUL";
+
   /* POSIX <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_13>
      specifies: "More than two leading <slash> characters shall be treated as
      a single <slash> character."  */
diff --git a/gettext-tools/gnulib-lib/stat.c b/gettext-tools/gnulib-lib/stat.c
index ebed63e..6663783 100644
--- a/gettext-tools/gnulib-lib/stat.c
+++ b/gettext-tools/gnulib-lib/stat.c
@@ -118,6 +118,10 @@ rpl_stat (char const *name, struct stat *buf)
      around length limitations
      <https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file> ?  */
 
+  /* To ease portability.  Like in open.c.  */
+  if (strcmp (name, "/dev/null") == 0)
+    name = "NUL";
+
   /* POSIX <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_13>
      specifies: "More than two leading <slash> characters shall be treated as
      a single <slash> character."  */
diff --git a/gettext-tools/libgettextpo/stat.c b/gettext-tools/libgettextpo/stat.c
index ebed63e..6663783 100644
--- a/gettext-tools/libgettextpo/stat.c
+++ b/gettext-tools/libgettextpo/stat.c
@@ -118,6 +118,10 @@ rpl_stat (char const *name, struct stat *buf)
      around length limitations
      <https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file> ?  */
 
+  /* To ease portability.  Like in open.c.  */
+  if (strcmp (name, "/dev/null") == 0)
+    name = "NUL";
+
   /* POSIX <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_13>
      specifies: "More than two leading <slash> characters shall be treated as
      a single <slash> character."  */
diff --git a/libtextstyle/lib/stat.c b/libtextstyle/lib/stat.c
index ebed63e..6663783 100644
--- a/libtextstyle/lib/stat.c
+++ b/libtextstyle/lib/stat.c
@@ -118,6 +118,10 @@ rpl_stat (char const *name, struct stat *buf)
      around length limitations
      <https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file> ?  */
 
+  /* To ease portability.  Like in open.c.  */
+  if (strcmp (name, "/dev/null") == 0)
+    name = "NUL";
+
   /* POSIX <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_13>
      specifies: "More than two leading <slash> characters shall be treated as
      a single <slash> character."  */
-- 
2.50.0.windows.2

