From: Bruno Haible <bruno@clisp.org>
Date: Fri, 13 Jun 2025 00:04:32 +0200
Subject: stat: Add support for mingw 13 in 32-bit mode

diff --git a/gettext-runtime/gnulib-lib/sys_stat.in.h b/gettext-runtime/gnulib-lib/sys_stat.in.h
index 8bde5a7..c3c38fd 100644
--- a/gettext-runtime/gnulib-lib/sys_stat.in.h
+++ b/gettext-runtime/gnulib-lib/sys_stat.in.h
@@ -849,7 +849,11 @@ _GL_WARN_ON_USE (mknodat, "mknodat is not portable - "
 #   elif @WINDOWS_64_BIT_ST_SIZE@
      /* Above, we define stat to _stati64.  */
 #    if defined __MINGW32__ && defined _stati64
-#     ifndef _USE_32BIT_TIME_T
+#     ifdef _USE_32BIT_TIME_T
+       /* The system headers possibly define _stati64 to _stat32i64.  */
+#      undef _stat32i64
+#      define _stat32i64(name, st) rpl_stat (name, st)
+#     else
        /* The system headers define _stati64 to _stat64.  */
 #      undef _stat64
 #      define _stat64(name, st) rpl_stat (name, st)
diff --git a/gettext-tools/gnulib-lib/sys_stat.in.h b/gettext-tools/gnulib-lib/sys_stat.in.h
index 8bde5a7..c3c38fd 100644
--- a/gettext-tools/gnulib-lib/sys_stat.in.h
+++ b/gettext-tools/gnulib-lib/sys_stat.in.h
@@ -849,7 +849,11 @@ _GL_WARN_ON_USE (mknodat, "mknodat is not portable - "
 #   elif @WINDOWS_64_BIT_ST_SIZE@
      /* Above, we define stat to _stati64.  */
 #    if defined __MINGW32__ && defined _stati64
-#     ifndef _USE_32BIT_TIME_T
+#     ifdef _USE_32BIT_TIME_T
+       /* The system headers possibly define _stati64 to _stat32i64.  */
+#      undef _stat32i64
+#      define _stat32i64(name, st) rpl_stat (name, st)
+#     else
        /* The system headers define _stati64 to _stat64.  */
 #      undef _stat64
 #      define _stat64(name, st) rpl_stat (name, st)
diff --git a/gettext-tools/libgettextpo/sys_stat.in.h b/gettext-tools/libgettextpo/sys_stat.in.h
index 8bde5a7..c3c38fd 100644
--- a/gettext-tools/libgettextpo/sys_stat.in.h
+++ b/gettext-tools/libgettextpo/sys_stat.in.h
@@ -849,7 +849,11 @@ _GL_WARN_ON_USE (mknodat, "mknodat is not portable - "
 #   elif @WINDOWS_64_BIT_ST_SIZE@
      /* Above, we define stat to _stati64.  */
 #    if defined __MINGW32__ && defined _stati64
-#     ifndef _USE_32BIT_TIME_T
+#     ifdef _USE_32BIT_TIME_T
+       /* The system headers possibly define _stati64 to _stat32i64.  */
+#      undef _stat32i64
+#      define _stat32i64(name, st) rpl_stat (name, st)
+#     else
        /* The system headers define _stati64 to _stat64.  */
 #      undef _stat64
 #      define _stat64(name, st) rpl_stat (name, st)
diff --git a/libtextstyle/lib/sys_stat.in.h b/libtextstyle/lib/sys_stat.in.h
index 8bde5a7..c3c38fd 100644
--- a/libtextstyle/lib/sys_stat.in.h
+++ b/libtextstyle/lib/sys_stat.in.h
@@ -849,7 +849,11 @@ _GL_WARN_ON_USE (mknodat, "mknodat is not portable - "
 #   elif @WINDOWS_64_BIT_ST_SIZE@
      /* Above, we define stat to _stati64.  */
 #    if defined __MINGW32__ && defined _stati64
-#     ifndef _USE_32BIT_TIME_T
+#     ifdef _USE_32BIT_TIME_T
+       /* The system headers possibly define _stati64 to _stat32i64.  */
+#      undef _stat32i64
+#      define _stat32i64(name, st) rpl_stat (name, st)
+#     else
        /* The system headers define _stati64 to _stat64.  */
 #      undef _stat64
 #      define _stat64(name, st) rpl_stat (name, st)
-- 
2.50.0.windows.2

