From: Michele Locati <michele@locati.it>
Date: Mon, 15 Apr 2019 18:10:02 +0200
Subject: [PATCH 1/5] Fix msginit

---
 gettext-tools/src/msginit.c | 48 -------------------------------------
 1 file changed, 48 deletions(-)

--- a/gettext-tools/src/msginit.c
+++ b/gettext-tools/src/msginit.c
@@ -874,54 +874,6 @@ project_id (const char *header)
       return old_field;
     }
 
-  gettextlibdir = getenv ("GETTEXTLIBDIR_SRCDIR");
-  if (gettextlibdir == NULL || gettextlibdir[0] == '\0')
-    gettextlibdir = relocate (LIBDIR "/gettext");
-
-  prog = xconcatenated_filename (gettextlibdir, "project-id", NULL);
-
-  /* Call the project-id shell script.  */
-  argv[0] = "/bin/sh";
-  argv[1] = prog;
-  argv[2] = NULL;
-  child = create_pipe_in (prog, "/bin/sh", argv, DEV_NULL, false, true, false,
-                          fd);
-  if (child == -1)
-    goto failed;
-
-  /* Retrieve its result.  */
-  fp = fdopen (fd[0], "r");
-  if (fp == NULL)
-    {
-      error (0, errno, _("fdopen() failed"));
-      goto failed;
-    }
-
-  line = NULL; linesize = 0;
-  linelen = getline (&line, &linesize, fp);
-  if (linelen == (size_t)(-1))
-    {
-      error (0, 0, _("%s subprocess I/O error"), prog);
-      fclose (fp);
-      goto failed;
-    }
-  if (linelen > 0 && line[linelen - 1] == '\n')
-    line[linelen - 1] = '\0';
-
-  fclose (fp);
-
-  /* Remove zombie process from process list, and retrieve exit status.  */
-  exitstatus = wait_subprocess (child, prog, false, false, true, false, NULL);
-  if (exitstatus != 0)
-    {
-      error (0, 0, _("%s subprocess failed with exit code %d"),
-             prog, exitstatus);
-      goto failed;
-    }
-
-  return line;
-
-failed:
   return "PACKAGE";
 }
 
-- 
2.20.1.windows.1

