From: Michele Locati <michele@locati.it>
Date: Mon, 29 Apr 2019 14:13:03 +0200
Subject: [PATCH] Move cldr-plurals

--- a/gettext-tools/src/msginit.c
+++ b/gettext-tools/src/msginit.c
@@ -1350,7 +1350,7 @@ plural_forms ()
   gettextcldrdir = getenv ("GETTEXTCLDRDIR");
   if (gettextcldrdir != NULL && gettextcldrdir[0] != '\0')
     {
-      const char *gettextlibdir;
+      char *gettextbindir;
       char *dirs[3], *last_dir;
       char *argv[4];
       pid_t child;
@@ -1361,11 +1361,18 @@ plural_forms ()
       size_t linelen;
       int exitstatus;
 
-      gettextlibdir = getenv ("GETTEXTLIBDIR_BUILDDIR");
-      if (gettextlibdir == NULL || gettextlibdir[0] == '\0')
-        gettextlibdir = relocate (LIBDIR "/gettext");
+      gettextbindir = xstrdup(program_name);
+      for (i = strlen(gettextbindir) - 1; i >= 0; i--)
+        {
+          if (gettextbindir[i] == '/' || gettextbindir[i] == '\\')
+            {
+              gettextbindir[i + 1] = '\0';
+              break;
+            }
+        }
 
-      prog = xconcatenated_filename (gettextlibdir, "cldr-plurals", NULL);
+      prog = xconcatenated_filename (gettextbindir, "cldr-plurals.exe", NULL);
+      free(gettextbindir);
 
       last_dir = xstrdup (gettextcldrdir);
       dirs[0] = "common";
-- 
2.20.1.windows.1
