From: Michele Locati <michele@locati.it>
Date: Tue, 30 Apr 2019 16:58:43 +0200
Subject: [PATCH] Fix for get_lcid (localename.c) required on Windows

--- a/gettext-runtime/intl/localename.c
+++ b/gettext-runtime/intl/localename.c
@@ -2604,6 +2604,10 @@ get_lcid (const char *locale_name)
   static LCID last_lcid;
   static char last_locale[1000];
 
+  if (locale_name == NULL)
+    {
+      return 0;
+    }
   /* Lock while looking for an LCID, to protect access to static
      variables: last_lcid, last_locale, found_lcid, and lname.  */
   gl_lock_lock (get_lcid_lock);
@@ -3252,7 +3256,7 @@ gl_locale_name_posix (int category, const char *categoryname)
         separated list of locales.  We need only one, so we take the
         one corresponding to LC_CTYPE, as the most important for
         character translations.  */
-      if (category == LC_ALL && strchr (locname, ';'))
+      if (category == LC_ALL && locname != NULL && strchr (locname, ';'))
         locname = setlocale (LC_CTYPE, NULL);
 
       /* Convert locale name to LCID.  We don't want to use
-- 
2.20.1.windows.1
